src/kOS/Screen/Interpreter.cs

```diff
@@ -157,6 +157,12 @@ protected void CompileCommand(string commandText)
             }
         }
 
+        public bool isWaitingForCommand()
+        {
+            IProgramContext context = ((CPU)Shared.Cpu).GetInterpreterContext();
+            return context.Program[context.InstructionPointer] is OpcodeEOF;
+        }
+
```

Comment:
I just pushed a new commit that fixed the above so it does the more elegant solution mentioned.